polyfill/lib/ecmascript.mjs

```diff
@@ -3528,20 +3528,15 @@ export const ES = ObjectAssign({}, ES2020, {
         relativeTo = yearsLater;
         days += monthsWeeksInDays;
 
-        // Years may be different lengths of days depending on the calendar, so
-        // we need to convert days to years in a loop. We get the number of days
-        // in the one-year period after (or preceding, depending on the sign of
-        // the duration) the relativeTo date, and convert that number of days to
-        // one year, repeating until the number of days is less than a year.
-        const sign = MathSign(days);
-        const oneYear = new TemporalDuration(days < 0 ? -1 : 1);
-        let oneYearDays;
-        ({ relativeTo, days: oneYearDays } = ES.MoveRelativeDate(calendar, relativeTo, oneYear));
-        while (MathAbs(days) >= MathAbs(oneYearDays)) {
-          years += sign;
-          days -= oneYearDays;
-          ({ relativeTo, days: oneYearDays } = ES.MoveRelativeDate(calendar, relativeTo, oneYear));
-        }
+        const daysLater = calendar.dateAdd(relativeTo, { days }, {}, TemporalDate);
+        const yearsPassed = calendar.dateUntil(relativeTo, daysLater, { largestUnit: 'years' }).years;
+        years += yearsPassed;
+        const oldRelativeTo = relativeTo;
+        relativeTo = calendar.dateAdd(relativeTo, { years: yearsPassed }, {}, TemporalDate);
+        const daysPassed = ES.DaysUntil(oldRelativeTo, relativeTo);
+        days -= daysPassed;
+        let { days: oneYearDays } = ES.MoveRelativeDate(calendar, relativeTo, new TemporalDuration(days < 0 ? -1 : 1));
```

Comment:
Maybe keep `new TemporalDuration(days < 0 ? -1 : 1)` as a constant named `oneYear` even though it's no longer used more than once? I'd find it easier to decipher.